cvmfs/sync_mediator.cc

```diff
@@ -28,6 +29,10 @@
 
 using namespace std;  // NOLINT
 
+namespace catalog {
+class DirectoryEntry;
```

Comment:
Since we need the full definition of `DirectoryEntry` anyway, we should include `directory_entry.h` instead of the forward declaration.